mysql/provider.go

```diff
@@ -171,6 +183,26 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 		if err != nil {
 			return nil, diag.Errorf("failed to register driver %v", err)
 		}
+	} else if d.Get("use_azure_ad").(bool) {
```

Comment:
It may be not the best programming, but we already use `cloudsql://` prefix fox cloudsql. Could we have something similar here? This is mostly for consistency. If we wanted to do a larger refactoring, I'd prefer enum with password provider - booleans don't allow for such extensibility. 

If it's difficult (maybe even considering your possible next step), then please explain and we may agree on this.

Reply:
I have no problem to switch `use_azure_ad` to any scheme prefix of endpoint, but it will be just scheme prefix working as this boolean.

`cloudsql` currently supports (thanks to Google CloudSql SDK) specifying database by GCP resource name instead of hostname AND then use Google's IAM access tokens.

I would suggest `azure` will support only Azure AD access tokens and not resource name -> host name resolution, as it would need more Azure SDKs to be imported (currently only azidentity is imported).

I'l update code this way in few moments.

Reply:
Thanks, this looks good!